### ISSUE TITLE
How do I pass multiple arguments through Keras Tuner Function?

### ISSUE THREAD

**Issue by umtek12** [keras-team/keras-tuner#402]
I am having difficulty figuring out how to pass multiple arguments through keras tuner function. I looked all over all available documentation and questions related to this and I could not find anything for this particular problem.

I just want to be able to pass additional arguments through this function:
`def build_model(hp, some_val_1, some_val_2)`

Overall Code (Simplified):
```
import kerastuner as kt

def build_model(hp, some_val_1, some_val_2):
    print(some_val_1)
    print(some_val_2)
    
    conv1d_val_1 = hp.Int("1-input_units", min_value=32, max_value=1028, step=64)
    conv1d_filt_1 = hp.Int("1b-filter_units", min_value=2, max_value=10, step=1)
    model.add(Conv1D(conv1d_val_1, conv1d_filt_1, activation='relu', input_shape=input_shape, padding='SAME'))
    model.add(Dense(1))
    model.compile(loss='mae', optimizer='adam')
    return model

model = kt.Hyperband(build_model, objective="val_loss", max_epochs = 10, factor = 3, directory=os.path.normpath(path_save_dir))
model.search(x=x_train, y=y_train, epochs=10, batch_size=500, validation_data=(x_test, y_test), shuffle=True)
```
Attempt 1 (I tried many variations) - Does not work:
`model = kt.Hyperband(build_model(kt.HyperParameters(), some_val_1, some_val_2), objective="val_loss", max_epochs = 10, factor = 3, directory=os.path.normpath(path_save_dir))`

Attempt 2 (I tried many variations) - Does not work:
`model = kt.Hyperband(build_model, some_val_1='1', some_val_2='2',objective="val_loss", max_epochs = 10, factor = 3, directory=os.path.normpath(path_save_dir))`

Attempt 3 (I tried many variations) - Does not work:
`model = kt.Hyperband(build_model, args=(some_val_1, some_val_2,),objective="val_loss", max_epochs = 10, factor = 3, directory=os.path.normpath(path_save_dir))`

Please send help

**Comment by overg123** [comment]
I think that what is missing is that you need to create an object.
You just created a class but didn't create an object.
Also the tuner return tuner, not a model. ( Although it is just a name still you need to understand the concept to not mix it up)

Try this:

**model_object**  = build_model ( all paramters.)
....

tuner = kt.tuners.RandomSearch(**model_object**....

**Comment by allard-jeff** [comment]
This should give an example of creating a class and initializing with some_value_1 etc: https://github.com/JulieProst/keras-tuner-tutorial/blob/master/hypermodels.py

You can also use fixed hyper parameters in your setup:
e.g. hp.Fixed('some_value_1', 5)
I think it is easier to create these outside of the model build function and pass in hp.

**Comment by yixingfu** [comment]
One thing you can do is 
```
from functools import partial

def build_model_with_extra_args(hp, some_val_1, some_val_2):
    ...

build_model = partial(build_model_with_extra_args, hp, some_val_1=value_for_some_val_1, some_val_2=value_for_some_val_2)
```

In Keras Tuner, you always need to pass a HyperModel object or a function that takes *only* an object in class hyperparameters (often called `hp` in the code) as argument.

**Comment by akashsindhu96** [comment]
Correction in the above answer.
```

from functools import partial

def build_model_with_extra_args(some_val_1, some_val_2, hp):
    ...

build_model = partial(build_model_with_extra_args, some_val_1=value_for_some_val_1, some_val_2=value_for_some_val_2)
```

**Comment by DEELMU-TECH** [comment]
Multi-objective using either KerasTuner Randomsearch, BaeysianOptimitation or Hyperband this is the appropriate syntax. i am putting it here because i have scoured the internet and there is no example of using a list of objectives using any of the tuner classes in keras tuner:
tuner = kerastuner.RandomSearch(model_builder,
                                                     objective=(["val_accuracy","val_loss"]),
                                                     max_trials=10,
                                                     seed=None,
                                                    hyperparameters=None,
                                                    tune_new_entries=True,
                                                   allow_new_entries=True,
                                                   )

**Comment by LPetesic** [comment]
you can use a lambda expression:

tuner =kt.Hyperband(
lambda hp: build_model(hp, some_val_1, some_val_2), 
...
)

**Comment by EnriqueGautoSand** [comment]
any way to tune the input shape?
in this case you are running several times passing it differents some_val_1
i found this code https://stackoverflow.com/questions/72710386/how-to-hypertune-input-shape-using-keras-tuner/72896040#72896040 that tune it but tried and not work
